lib/vat.js

```diff
@@ -36,17 +40,24 @@ function validate(data, schema) {
       let allKeys = _.union(Object.keys(data), Object.keys(schema));
       value = {};
       allKeys.forEach(function (key) {
-        let result = validate(data[key], schema[key]);
-
-        if (result.error) {
-          result.error.key = key;
-          throw result.error;
-        }
-
-        let targetKey = schema[key].renameTo() || key;
-        // only add undefined values iff the value is required
-        if (! _.isUndefined(result.value) || schema[key]._isRequired) {
-          value[targetKey] = result.value;
+        if (schema[key]) {
+          let result = validate(data[key], schema[key], options);
```

Comment:
Could you use `const` for `result` and `targetKey` below? Was under the impression that const was ok as long as you are not reassigning the variable name.